tests/registry/test_predicate_mapping.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+
+from eth_abi.registry import (
+    PredicateMapping,
+)
+
+
+@pytest.fixture
+def mapping():
+    return PredicateMapping()
+
+
+def gt(x, y):
+    return x > y
+
+
+def lte(x, y):
+    return x <= y
+
+
+def test_can_add_predicates_and_find_corresponding_values(mapping):
+    mapping.add(gt, 'greater than')
+    mapping.add(lte, 'less than or equal')
+
+    assert mapping.find(2, 1) == 'greater than'
+    assert mapping.find(2, 2) == 'less than or equal'
+    assert mapping.find(2, 3) == 'less than or equal'
+
+
+def test_cannot_add_same_predicate_twice(mapping):
+    mapping.add(gt, 'greater')
+
+    with pytest.raises(ValueError) as e:
+        mapping.add(gt, 'something else')
+
+    e.match('Predicate already exists')
+
+
+def test_cannot_add_two_predicates_with_same_label(mapping):
+    mapping.add(gt, 'foo', 'my_label')
+
+    with pytest.raises(ValueError) as e:
+        mapping.add(lte, 'bar', 'my_label')
+
+    e.match('Predicate with label')
```

Comment:
Are you familiar with the `match` kwarg that `pytest.raises` accepts?

https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

It should allow this to be written more succinctly as:

```python
with pytest.raises(ValueError, match="Predicate with label * already exists"):
    mapping.add(lte, 'bar', 'my_label')
```

Reply:
I wasn't familiar with that.  I haven't used pytest extensively so I'm happy to hear about any cool bells and whistles.